validator_client/src/http_api/mod.rs

```diff
@@ -398,6 +412,33 @@ pub fn serve<T: 'static + SlotClock + Clone, E: EthSpec>(
             },
         );
 
+    // GET lighthouse/ui/fallback_health
+    let get_lighthouse_ui_fallback_health = warp::path("lighthouse")
+        .and(warp::path("ui"))
+        .and(warp::path("fallback_health"))
+        .and(warp::path::end())
+        .and(block_service_filter.clone())
+        .then(|block_filter: BlockService<T, E>| async move {
+            let mut result: HashMap<(usize, String), Result<BeaconNodeHealth, CandidateError>> =
+                HashMap::new();
+            for node in &*block_filter.beacon_nodes.candidates.read().await {
+                result.insert(
+                    (node.index, node.beacon_node.to_string()),
+                    *node.health.read(),
+                );
+            }
+            if let Some(proposer_nodes) = &block_filter.proposer_nodes {
+                for node in &*proposer_nodes.candidates.read().await {
+                    result.insert(
+                        (node.index, node.beacon_node.to_string()),
+                        *node.health.read(),
+                    );
+                }
+            }
+
+            blocking_json_task(move || Ok(api_types::GenericResponse::from(result))).await
```

Comment:
I wrote a little test to see how the response look like and I'm not able to get the response to serialize, it looks like the key of the `HashMap` must be string?

```rust
    #[test]
    fn json_response() {
        let results = vec![
            Ok(BeaconNodeHealth {
                user_index: 0,
                head: Slot::new(100),
                optimistic_status: IsOptimistic::Yes,
                execution_status: ExecutionEngineHealth::Healthy,
                health_tier: BeaconNodeHealthTier {
                    tier: 1,
                    sync_distance: SyncDistance::new(0),
                    distance_tier: SyncDistanceTier::Synced,
                },
            }),
            Err(CandidateError::Incompatible),
        ];
        let mut result: HashMap<(usize, String), Result<BeaconNodeHealth, CandidateError>> =
            HashMap::new();
        result.insert((0, "http://localhost:5052".to_string()), results[0]);
        result.insert((1, "http://localhost:5053".to_string()), results[1]);
        let res = serde_json::to_string(&GenericResponse::from(result)).unwrap();
        println!("{}", res);
    }
```

if I remove the index, it looks like this: 

```json
{
  "data": {
    "http://localhost:5053": {
      "Err": "Incompatible"
    },
    "http://localhost:5052": {
      "Ok": {
        "user_index": 0,
        "head": "100",
        "optimistic_status": "Yes",
        "execution_status": "Healthy",
        "health_tier": {
          "tier": 1,
          "sync_distance": "0",
          "distance_tier": "Synced"
        }
      }
    }
  }
}
```

Reply:
What about something like this?

This structure differentiate between proposer and regular beacon node, and also preserve the order of them as well:

I also noticed that if we don't get health data from the BN, we don't return the `user_index` of the node. Maybe we can lift that field out of the health struct?

```json
{
  "data": {
    "proposers": [
      {
        "url": "http://localhost:5053",
        "error": "Incompatible"
      },
      {
        "url": "http://localhost:5052",
        "health": {
          "user_index": 0,
          "head": "100",
          "optimistic_status": "Yes",
          "execution_status": "Healthy",
          "health_tier": {
            "tier": 1,
            "sync_distance": "0",
            "distance_tier": "Synced"
          }
        }
      }
    ],
    "beacon_nodes": [
      {
        "url": "http://localhost:5053",
        "error": "Incompatible"
      },
      {
        "url": "http://localhost:5052",
        "health": {
          "user_index": 0,
          "head": "100",
          "optimistic_status": "Yes",
          "execution_status": "Healthy",
          "health_tier": {
            "tier": 1,
            "sync_distance": "0",
            "distance_tier": "Synced"
          }
        }
      }
    ]
  }
}
```

Reply:
Good suggestion. This is now implemented in 
5cc2c13.

I have added the user index into the top level (but it is duplicated in the health when that is available. I decided that the extra complexity in removing it wasn't worth doing).

I also added a custom `Serialize` impl to get the exact behaviour in your example (where health/error is dependent on the `Result` enum. (Although this does mean that it has incompatible deserialization, but I don't think there should ever be a case where we need to Deserialize?)